ProcessMaker/Traits/HasAuthorization.php

```diff
@@ -42,12 +42,8 @@ public function loadGroupPermissions()
     public function hasPermission($permissionString)
     {
         if (\Auth::user() == $this) {
-            if (session('permissions')) {
-                $permissionStrings = session('permissions');
-            } else {
-                $permissionStrings = $this->loadPermissions();
-                session(['permissions' => $permissionStrings]);
-            }
+            $permissionStrings = $this->loadPermissions();
```

Comment:
Take in count that the `->hasPermission()` can be called multiple times in a same block of code (e.g. controller middleware, menus). Please consider to load the permissions only once per User Object Instance.

Reply:
I agree, I have added a block of code in auth middleware to load all user permissions in each request